CRM/CiviGeometry/BAO/Geometry.php

```diff
@@ -88,33 +88,27 @@ public static function getCentroid($params) {
   public static function contains($params) {
     $multipleResult = [];
     $dualIntegerSQL = "SELECT a.id, ST_Contains(a.geometry, b.geometry) as contains_result
-      FROM civigeometry_geometry a, civigeometry_geometry b
-      WHERE b.id = %2";
-    $singleIntergerSQL = "SELECT id, ST_Contains(geometry, GeomFromText(%1, 4326)) as contains_result
-      FROM civigeometry_geometry";
+        FROM civigeometry_geometry a, civigeometry_geometry b";
+    $dualIntegerWhere = " WHERE b.id = %2 AND a.is_archived = 0 AND b.is_archived = 0";
+    $singleIntegerSQL = "SELECT cg.id, ST_Contains(cg.geometry, GeomFromText(%1, 4326)) as contains_result
+      FROM civigeometry_geometry cg";
+    $singleIntegerWhere = " WHERE cg.is_archived = 0";
     if ($params['geometry_a'] == 0) {
-      $geometryParams = [
-        'is_archived' => 0,
-        'options' => ['limit' => 0],
-        'return' => ['id'],
-      ];
-      if (!empty($params['geometry_a_collection_id'])) {
-        $geometryParams['collection_id'] = $params['geometry_a_collection_id'];
-      }
-      $geometries = civicrm_api3('Geometry', 'get', $geometryParams)['values'];
-      $singleIntergerSQL .= " WHERE id IN (%2)";
-      $dualIntegerSQL .= " AND a.id IN (%1)";
+      $singleIntegerSQL .= "INNER JOIN civigeometry_collection_geometry cgc ON cgc.geometry_id = cg.id";
```

Comment:
Preceding space required?

Reply:
Thanks your right added it in now @andrew-cormick-dockery